### PR TITLE
Split up fiat-crypto CI into two targets

### PR DESCRIPTION
**Pull request by JasonGross**
There is the new pipeline, and the old pipeline.  Most of what they
share in common is the (very large) library of lemmas about `Z`.

As per the discussion in
https://github.com/coq/coq/pull/8064#issuecomment-413474176 through
https://github.com/coq/coq/pull/8064#issuecomment-413793143

**Kind:** performance / infrastructure.

Two things should happen before this is merged:
 - [x] We should make sure that the timings of the two parts are reasonable on the CI before merging this.
  + It looks like the times are 117 minutes 28 seconds for the old pipeline + spec stuff; and 39 minutes 44 seconds for the new pipeline
 - [x] I should make sure that the targets are actually building the right things by inspecting the CI logs.

### CODE REVIEW

File: dev/ci/ci-basic-overlay.sh [hunk 1]
```diff
@@ -104,6 +104,12 @@
 : "${fiat_parsers_CI_BRANCH:=master}"
 : "${fiat_parsers_CI_GITURL:=https://github.com/mit-plv/fiat}"
 
+########################################################################
+# fiat_crypto_legacy
+########################################################################
+: "${fiat_crypto_legacy_CI_BRANCH:=master}"
+: "${fiat_crypto_legacy_CI_GITURL:=https://github.com/mit-plv/fiat-crypto}"
```
[the remark this answers is not in the document]
Comment (JasonGross) [reply dev/ci/ci-basic-overlay.sh]: Yes, though at some point we might split off the old pipeline into a separate branch or repo.  Would you prefer a single pair of variables for now?

[the remark this answers is not in the document]
Comment (SkySkimmer) [reply dev/ci/ci-basic-overlay.sh]: Until they're really separate it just makes overlays more annoying to make, so let's keep them together.

**Review by Zimmi48 — COMMENTED** [review]
Thanks!
Shouldn't we also have the common intermediate target get a `ci-fiat-crypto-common` target on which both other targets get to depend, so that once https://gitlab.com/gitlab-org/gitlab-ce/issues/41947 lands we can share this part of the build?

File: dev/ci/ci-basic-overlay.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (JasonGross) [reply dev/ci/ci-basic-overlay.sh]: Done

**Comment by JasonGross** [comment]
> Shouldn't we also have the common intermediate target get a ci-fiat-crypto-common target 

I've just pushed an `util` target to fiat-crypto, which is an approximation of the common overlap.  I don't actually know the CI infrastructure well enough to be confident about how to make shared targets.  Are you asking just for a `.sh` file that both of these files source, or a `.sh` file together with a line in `Makefile.ci`, or something else?

**Comment by JasonGross** [comment]
However, note that the util target takes only about 4 minutes of user time to build on my machine.  The timing profile (truncated to exclude files that take less than 5 seconds) is:
```
Time     | File Name
---------------------------------------------------
4m07.50s | Total
---------------------------------------------------
0m24.73s | bbv/theories/Word
0m16.38s | Util/ZUtil/LandLorBounds
0m14.98s | Algebra/Field
0m12.03s | Algebra/Ring
0m10.01s | Util/ZUtil/ZSimplify/Autogenerated
0m09.27s | Util/ZRange/CornersMonotoneBounds
0m08.54s | Util/ListUtil
0m07.61s | Util/ZUtil/Modulo
0m06.11s | Util/FixedWordSizesEquality
0m06.00s | Util/ZUtil/Morphisms
0m05.35s | Util/FsatzAutoLemmas
```
So it's not clear to me that we'd get much speed-up by having a fiat-crypto common target (and if the network / boot is slow, it might even make things slower overall).

**Comment by Zimmi48** [comment]
A `.sh` file together with a line in `Makefile.ci` and a dependency declaration in `Makefile.ci` like these ones:

https://github.com/coq/coq/blob/05d13bf516a75da38bacbb62608528f8dceb34fd/Makefile.ci#L50-L58

But it doesn't necessarily make sense unless the dependency is installed before the rest is compiled... @SkySkimmer?

**Comment by Zimmi48** [comment]
>So it's not clear to me that we'd get much speed-up by having a fiat-crypto common target (and if the network / boot is slow, it might even make things slower overall).

Yeah, it probably make sense to keep things as they are I guess.

**Review by SkySkimmer — APPROVED** [review]

**Comment by Zimmi48** [comment]
I think @SkySkimmer should feel free to proceed and merge this. The fact that @ejgallego is still the owner for `Makefile.ci` (instead of the CI team) was an overlook from me, that is fixed in #8323.

**Pull request merged**